mqtt_client/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@ set(ROS_VERSION $ENV{ROS_VERSION})
 
 ## Compile as C++17
 add_compile_options(-std=c++17)
+link_libraries("$<$<AND:$<CXX_COMPILER_ID:GNU>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.0>>:-lstdc++fs>")
```

Comment:
Thanks a lot! I suggest to move this to the existing `target_link_libraries` commands. Note that there are two such lines, one for ROS, one for ROS 2.